src/network/webtask.cpp

```diff
@@ -317,22 +255,35 @@ void WebTask::slotNetworkReplyFinished() {
         return;
     }
 
-    VERIFY_OR_DEBUG_ASSERT(m_state == State::Pending) {
-        DEBUG_ASSERT(m_timeoutTimerId == kInvalidTimerId);
-        return;
-    }
-
     if (m_timeoutTimerId != kInvalidTimerId) {
         killTimer(m_timeoutTimerId);
         m_timeoutTimerId = kInvalidTimerId;
     }
 
+    if (m_state != State::Pending) {
+        kLogger.debug()
+                << this
+                << "Discarding obsolete network reply"
+                << pFinishedNetworkReply;
+        if (m_state == State::Aborting) {
+            const auto requestUrl = pPendingNetworkReply->request().url();
+            emitAborted(requestUrl);
+        } else {
+            // Might have been aborted or timed out in the meantime
```

Comment:
How about the caller in this case is he informed, or should we signal it just in case?

Reply:
Nope. We are already in a final state and a signal has already been emitted. Otherwise you would two notifications for a single request which is unexpected. This special case might only occur if a delayed network response has already been enqueued when the client aborts the request.

Reply:
I will add a comment.